### PR TITLE
Ensuring that there are at most 5 security groups per instance (The max is configurable)

### PR DESCRIPTION
**Pull request by jeyrschabu**
- added a check to guard against deploying a server group with more than 5 SGs per instance in VPC

### CODE REVIEW

**Comment by jeyrschabu** [comment]
Going to look in DeployDefaults to get limits. This isn't ready for a merge

**Comment by jeyrschabu** [comment]
@cfieber @anotherchrisberry PTAL

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy [hunk 1]
```diff
@@ -203,6 +204,10 @@ class AutoScalingWorker {
     // Favor subnetIds over availability zones
     def subnetIds = subnetIds?.join(',')
     if (subnetIds) {
+      if (securityGroups && maxSecurityGroupsPerInstance && securityGroups.size() > maxSecurityGroupsPerInstance) {
+        throw new IllegalArgumentException("An instance can have at most ${maxSecurityGroupsPerInstance} security groups")
+      }
```
Comment (anotherchrisberry) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy]: Does this limit only apply to VPC instances?

Comment (jeyrschabu) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy]: yeah. we know it's VPC if we have subnets

Comment (jeyrschabu) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy]: Only applies in VPC

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy [hunk 2]
```diff
@@ -81,6 +81,7 @@ class AutoScalingWorker {
   private int desiredInstances
 
   private RegionScopedProviderFactory.RegionScopedProvider regionScopedProvider
+  private Integer maxSecurityGroupsPerInstance
```
Comment (anotherchrisberry) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorker.groovy]: Any reason to make this configurable? Isn't it always 5?

**Comment by jeyrschabu** [comment]
merging this down

**Pull request merged**